File: src/main/java/io/beaniejoy/dongnecafe/cafe/dto/menu/OptionDetailResponseDto.java
package io.beaniejoy.dongnecafe.cafe.dto.menu;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class OptionDetailResponseDto {

    private Long optionDetailId;

    private String name;

    private Integer extra;
}

File: src/main/java/io/beaniejoy/dongnecafe/domain/cafe/dto/menu/OptionDetailResponseDto.kt
package io.beaniejoy.dongnecafe.domain.cafe.dto.menu

import io.beaniejoy.dongnecafe.domain.cafe.domain.OptionDetail
import java.math.BigDecimal

data class OptionDetailResponseDto(
    val id: Long = 0L,
    val name: String? = null,
    val extra: BigDecimal = BigDecimal.ZERO
) {
    companion object {
        fun of(optionDetail: OptionDetail): OptionDetailResponseDto {
            return OptionDetailResponseDto(
                id = optionDetail.id,
                name = optionDetail.name,
                extra = optionDetail.extra
            )
        }
    }
}